src/v/redpanda/admin_server.cc

```diff
@@ -3316,6 +3317,63 @@ void admin_server::register_self_test_routes() {
       });
 }
 
+ss::future<ss::json::json_return_type> admin_server::get_raft_metrics_handler(
+  std::unique_ptr<ss::httpd::request> req) {
+    const model::ntp ntp = parse_ntp_from_request(req->param);
+
+    if (need_redirect_to_leader(ntp, _metadata_cache)) {
+        throw co_await redirect_to_leader(*req, ntp);
+    }
+
+    seastar::httpd::debug_json::raft_metrics ans;
+
+    const bool is_controller = ntp == model::controller_ntp;
+    cluster::consensus_ptr raft;
+
+    if (!is_controller) {
+        auto partition = _partition_manager.local().get(ntp);
+        if (!partition) {
+            throw ss::httpd::not_found_exception(
+              fmt::format("Could not find ntp: {}", ntp));
+        }
+
+        raft = partition->raft();
+    } else {
+        raft = _controller->get_raft0();
+    }
+
+    if (!raft->is_leader()) {
+        throw ss::httpd::server_error_exception(
+          fmt_with_ctx(fmt::format, "Can not find leader for ntp {}", ntp));
+    }
+
+    seastar::httpd::debug_json::raft_metrics_leader leader_metrics;
+    leader_metrics.id = raft->get_leader_id().value_or(model::node_id(-1));
```

Comment:
is this line contradictory with `raft->is_leader` check?

Reply:
Yeah. I think just get value is ok